hyperspy/model.py

```diff
@@ -1480,7 +1483,11 @@ def print_current_values(self, only_free=True):
                 parameters = component.free_parameters if only_free \
                     else component.parameters
                 for parameter in parameters:
-                    if not hasattr(parameter.value, '__iter__'):
+                    if hasattr(parameter.value, '__iter__'):
+                        if skip_multi:
+                            for idx in range(len(parameter.value)):
+                                print("\t\ta[%d]\t%g" % (idx, parameter.value[idx]))
+                    else:
```

Comment:
No, this should be correct. The `else:` block concerns parameters that do not have the attribute `__iter__` set. The `if skip_multi:` is only applied to iterable parameters.